### PR TITLE
BYOC support without label

### PR DESCRIPTION
**Pull request by qpdpQ**
for ticket: https://github.ibm.com/IBMPrivateCloud/roadmap/issues/57312

It will check if `cs-ca-certificate-secret` is already there during the installation. If we find this secret but do not find the certificate, we will skip deploying `cs-ca-cert`

### CODE REVIEW

File: controllers/bootstrap/init.go [hunk 1]
```diff
@@ -953,7 +955,33 @@ func CheckClusterType(mgr manager.Manager, ns string) (bool, error) {
...
+func (b *Bootstrap) IsBYOCert() (bool, error) {
+	klog.V(2).Info("Detect if it is BYO cert")
+	secretName := "cs-ca-ceritifcate-secret"
+	secret := &corev1.Secret{}
+	err := b.Client.Get(context.TODO(), types.NamespacedName{Name: secretName, Namespace: b.CSData.ServicesNs}, secret)
+	if err != nil {
+		if !errors.IsNotFound(err) {
+			return false, err
+		}
+	}
```
Comment (bitscuit) [inline controllers/bootstrap/init.go:970]: What happens if error is `IsNotFound`? Secret will be empty, but there is no code to check if secret is empty or not before checking for certificate object.

File: controllers/bootstrap/init.go [hunk 2]
```diff
@@ -953,7 +955,33 @@ func CheckClusterType(mgr manager.Manager, ns string) (bool, error) {
...
+	certName := "cs-ca-certificate"
+	cert := &certmanagerv1.Certificate{}
+	certerr := b.Client.Get(context.TODO(), types.NamespacedName{Name: certName, Namespace: b.CSData.ServicesNs}, cert)
+	if certerr != nil {
+		if errors.IsNotFound(certerr) {
+			return true, nil
+		}
```
Comment (bitscuit) [inline controllers/bootstrap/init.go:977]: This is assuming that secret was found. But what if getting the secret returned `IsNotFound` error? This error is being ignored right now.

**Review by bitscuit — CHANGES_REQUESTED** [review]

File: controllers/bootstrap/init.go (hunk 1 above)
Comment (qpdpQ) [reply controllers/bootstrap/init.go:970]: Ah I didn't notice this, thanks for pointing it out

File: controllers/bootstrap/init.go (hunk 2 above)
Comment (qpdpQ) [reply controllers/bootstrap/init.go:978]: Yes, if we didn't get this secret, we should return false directly.

**Review by bitscuit — APPROVED** [review]
/lgtm

**Pull request merged**